astroid/raw_building.py

```diff
@@ -314,12 +330,14 @@ def inspect_build(
         self.object_build(node, module)
         return node
 
-    def object_build(self, node, obj):
+    def object_build(
+        self, node: nodes.Module | nodes.ClassDef, obj: types.ModuleType | type
+    ) -> None:
         """recursive method which create a partial ast from real objects
         (only function, class, and method are handled)
         """
         if obj in self._done:
-            return self._done[obj]
+            return None
```

Comment:
Could you explain the reasoning ?

Reply:
It's never used as a return value. The final `return` in this method just returns `None`. So this is an early exit check to see if we should build this object or if we have already done so.
For some reason it was decided to also return the already finished object, but that's not necessary for anything and unnecessarily complicates the typing/signature.